### PR TITLE
[Refactor Sessions] Fix getting cookie from session for waterbutler and archiving

### PR DESCRIPTION
**Pull request by cslzchen**
## Purpose

Fix session/cookie for WB/archiving

See https://centerforopenscience.slack.com/archives/C04KLR892GL/p1681926121439639 for details

## Changes

* Add `None` check for `UserSessionMap`
* Remove redundant `.save()` by updating session data before `.create()`

## QA Notes

Create registration
* No project: upload file during creation
* From project: make sure project has file

After submission, check you email to see that registration is created successfully (i.e no archiving errors)

## Documentation

N/A

## Side Effects

N/A

## Ticket

Bug fixes for for https://openscience.atlassian.net/browse/ENG-4316

### CODE REVIEW

File: osf/models/user.py [hunk 1]
```diff
@@ -1802,31 +1802,27 @@ def get_activity_points(self):
...
             user_session['auth_user_fullname'] = self.fullname
-            user_session.save()
+            user_session.create()
             UserSessionMap.objects.create(
                 user=self,
                 session_key=user_session.session_key,
                 expire_date=user_session.get_expiry_date()
             )
         else:
             user_session = SessionStore(session_key=user_session_map.session_key)
```
Comment (adlius) [inline osf/models/user.py:1825]: Maybe refactor this to something like:
```
user_session = SessionStore()
if user_session_map and SessionStore().exists(session_key=user_session_map.session_key):
        user_session = SessionStore(session_key=user_session_map.session_key)
    else:
        user_session.create()
        user_session['auth_user_id'] = self._id
        user_session['auth_user_username'] = self.username
        user_session['auth_user_fullname'] = self.fullname
        user_session.save()
        UserSessionMap.objects.create(
            user=self,
            session_key=user_session.session_key,
            expire_date=user_session.get_expiry_date()
        )
```

Comment (cslzchen) [reply osf/models/user.py:1825]: Two notes:
* Why defining `user_session = SessionStore()` if this is only used in the `else`
* And should we avoid `.create()`->setting data->`.save()` by calling `.create()` after setting data?

Comment (cslzchen) [reply osf/models/user.py:1825]: I still need to revert `if ... else ...` so that the condition is shorter and that `if ... else ...` looks better.

Comment (adlius) [reply osf/models/user.py:1825]: It's okay if we don't do `user_session = SessionStore()` and call `create()` after setting session data. But `not statement1 or not statement2` is a bit counter-intuitive for me.

**Pull request merged**